.github/workflows/cypress-test-multiauth-e2e.yml

```diff
@@ -0,0 +1,210 @@
+        shell: bash
+
+      # OSD bootstrap
+      - name: Run Dashboard with Security Dashboards Plugin
+        uses: ./.github/actions/install-dashboards
+        with:
+          plugin_name: security-dashboards-plugin
+
+      # Setup and Run SAML Idp
+      - name: Get and run SAML Idp on Linux
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          cd ./OpenSearch-Dashboards/plugins/security-dashboards-plugin
+          yarn pretest:jest_server
+
+      # Configure the Dashboard for SAML setup
+      - name: Configure and Run OpenSearch Dashboards with SAML Configuration
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          cd ./OpenSearch-Dashboards
+          rm -rf ./config/opensearch_dashboards.yml
+          cat << 'EOT' > ./config/opensearch_dashboards.yml
+          server.host: "localhost"
+          opensearch.hosts: ["https://localhost:9200"]
+          opensearch.ssl.verificationMode: none
+          opensearch.username: "kibanaserver"
+          opensearch.password: "kibanaserver"
+          opensearch.requestHeadersWhitelist: [ authorization,securitytenant ]
+          opensearch_security.multitenancy.enabled: true
+          opensearch_security.multitenancy.tenants.preferred: ["Private", "Global"]
+          opensearch_security.readonly_mode.roles: ["kibana_read_only"]
+          opensearch_security.cookie.secure: false
+          server.xsrf.allowlist: ["/_plugins/_security/api/authtoken", "/_opendistro/_security/api/authtoken", "/_opendistro/_security/saml/acs", "/_opendistro/_security/saml/acs/idpinitiated", "/_opendistro/_security/saml/logout"]
+          opensearch_security.auth.type: ["basicauth","saml"]
+          opensearch_security.auth.multiple_auth_enabled: true
+          opensearch_security.auth.anonymous_auth_enabled: false
+          home.disableWelcomeScreen: true
+          EOT
+          echo 'HERE IS THE DASHBOARD CONFIG FILE: '
+          cat ./config/opensearch_dashboards.yml
+          nohup yarn start --no-base-path --no-watch | tee dashboard.log &
+
+      # Check if OSD is ready with a max timeout of 600 seconds
+      - name : Check If OpenSearch Dashboards Is Ready
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          cd ./OpenSearch-Dashboards
+          echo "Start checking OpenSearch Dashboards."
+          for i in {1..60}; do
+            if grep -q "bundles compiled successfully after" "dashboard.log"; then
+              echo "OpenSearch Dashboards compiled successfully."
+              break
+            fi
+            if [ $i -eq 60 ]; then
+              echo "Timeout for 600 seconds reached. OpenSearch Dashboards did not finish compiling."
+              exit 1
+            fi
+            sleep 10
+          done
+
+      - name: Run Cypress
+        run : |
+          yarn add cypress --save-dev
+          yarn cypress:run --browser chrome --headless --env loginMethod=saml_multiauth --spec 'test/cypress/e2e/saml/*.js'
```

Comment:
This file is 99% the same as `cypress-test-saml-e2e.yml`, lets refactor all these cypress workflows together.  If there are only 2 lines different, lets figure out how to bake these together.

I suspect that by writing a way to append to the opensearch_dashboards.yml then its just a matter of defining a matrix of environments.

Reply:
This is refactored into a reusable action now.